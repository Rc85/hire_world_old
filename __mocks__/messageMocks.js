export const userMessage = {
    message_id: 1,
    belongs_to_job: 1,
    message_body: 'test',
    message_date: new Date(),
    message_type: 'User',
    message_sender: 'client',
    message_recipient: 'user',
    message_status: 'New',
    is_reply: 'false',
    message_modified_date: new Date()
}

export const systemUpdateMessage = Object.assign({}, userMessage);

systemUpdateMessage['message_type'] = 'Update';
systemUpdateMessage['message_sender'] = 'System';

export const systemConfirmMessage = Object.assign({}, userMessage);

systemConfirmMessage['message_type'] = 'Confirmation';
systemConfirmMessage['message_sender'] = 'System';