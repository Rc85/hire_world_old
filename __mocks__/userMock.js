export const User = {
    account_type: "Listing",
    allow_messaging: true,
    avatar_url: "/user_files/1/businessman-avatar-icon-01-.jpg",
    display_fullname: true,
    email_notifications: false,
    hide_email: false,
    user_address: "3475 E 27th Ave",
    user_bio: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Assumenda ullam repellat maiores aperiam debitis iure, repudiandae maxime pariatur nam rerum quasi numquam aliquid illo temporibus amet officiis, veritatis ipsum? Illum. Lorem ipsum dolor sit amet consectetur adipisicing elit. Aspernatur similique quae tenetur repellat earum voluptate, reiciendis doloribus. Pariatur atque porro sunt autem quod harum soluta necessitatibus tempora hic. Molestias, harum! Lorem ipsum dolor sit amet consectetur adipisicing elit. Hic molestiae nihil, optio accusantium repellat ex cupiditate commodi similique facere minus maxime explicabo dolorum neque veniam corrupti ipsa debitis. Pariatur, necessitatibus. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus voluptatem, a molestias aperiam obcaecati doloribus harum veritatis suscipit asperiores hic exercitationem molestiae aliquid soluta minima! Laborum placeat saepe dolores! Enim.",
    user_business_name: "RC Web Development",
    user_city: "Vancouver",
    user_city_code: "V5R1P8",
    user_country: "Canada",
    user_education: "BCIT",
    user_email: "rogerchin85@gmail.com",
    user_facebook: "https://www.facebook.com",
    user_firstname: "Roger",
    user_github: "https://github.com/Rc85",
    user_instagram: "https://www.instagram.com",
    user_last_login: "2018-12-10T00:32:16.540Z",
    user_lastname: "Chin",
    user_linkedin: "https://www.linkedin.com",
    user_phone: "604-889-8353",
    user_profile_id: 1,
    user_region: "British Columbia",
    user_setting_id: 1,
    user_title: "Web Developer",
    user_twitter: "https://www.twitter.com",
    user_website: "https://www.rogerchin.me",
    user_worldwide: null,
    username: "roger85"
}