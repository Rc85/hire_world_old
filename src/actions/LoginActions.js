import fetch from 'axios';
import { Alert } from '../actions/AlertActions';
import { LogError } from '../components/utils/LogError';
import { GetSession } from './FetchActions';

export const LoginUser = (data) => {
    return dispatch => {
        document.body.style.overflow = 'hidden';
        dispatch(LoginBegin('login begin'));

        return fetch.post('/api/auth/login', data)
        .then(resp => {
            document.body.style.overflow = 'auto';
            if (resp.data.status === 'success') {
                dispatch(LoginSuccess(resp.data.status, resp.data.user));
            } else if (resp.data.status === 'error') {
                dispatch(LoginError(resp.data.status, resp.data.statusMessage));
                dispatch(Alert(resp.data.status, resp.data.statusMessage));
            } else if (resp.data.status === 'access error') {
                dispatch(LoginError(resp.data.status, resp.data.statusMessage));
            }
        })
        .catch(err => LogError(err, '/api/auth/login'));   
    }
}

const LoginBegin = (status) => {
    return {
        type: 'LOGIN_USER',
        status
    }
}

const LoginSuccess = (status, user) => {
    return {
        type: 'LOGIN_USER_UPDATE',
        user,
        status
    }
}

const LoginError = (status, statusMessage) => {
    return {
        type: 'LOGIN_USER_ERROR',
        status,
        statusMessage, 
        user: null
    }
}

export const LogoutUser = () => {
    return dispatch => {
        fetch.post('/api/auth/logout')
        .then(resp => {
            if (resp.data.status === 'error') {
                dispatch(Logout('not logged in'));
            }
        })
        .catch(err => LogError(err, '/api/auth/logout'));
    }
}

const Logout = (status) => {
    return {
        type: 'LOGOUT_USER',
        status
    }
}

export const UpdateUser = user => {
    return {
        type: 'USER_UPDATE',
        user
    }
}

export const ResetLogin = () => {
    return {
        type: 'LOGIN_USER_ERROR',
        status: null,
        statusMessage: null
    }
}