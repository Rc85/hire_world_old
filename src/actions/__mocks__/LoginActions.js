export const User = {
    avatar_url: "../images/profile.png",
    business_name: "",
    default_location: false,
    display_business_name: false,
    display_contacts: false,
    display_fullname: false,
    email_notifications: true,
    hide_email: false,
    services_allowed: 3,
    user_address: null,
    user_bio: null,
    user_city: "",
    user_country: "",
    user_created_on: "2018-08-30T00:20:31.456Z",
    user_education: null,
    user_email: "hacker@hacker.com",
    user_facebook: null,
    user_firstname: "",
    user_github: null,
    user_id: 19,
    user_instagram: null,
    user_lastname: "",
    user_level: 0,
    user_linkedin: null,
    user_phone: null,
    user_region: "",
    user_status: "Active",
    user_title: null,
    user_twitter: null,
    user_type: null,
    user_website: null,
    user_worldwide: false,
    username: "hacker",
    view_counter: 0
}