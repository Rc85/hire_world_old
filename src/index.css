@font-face {
    font-family: 'Montserrat';
    src: url('./fonts/Montserrat-Regular.ttf');
}

@font-face {
    font-family: 'Montserrat Bold';
    src: url('./fonts/Montserrat-Bold.ttf');
}

html {
    min-height: 100vh;
}

body {
    margin: 0;
    padding: 0;
    font-family: 'Montserrat';
    background: #C4C4C4;
    color: #F9F6F4;
    min-height: 100vh;
    display: flex;
    overflow-x: hidden;
}

#root {
    width: 100%;
    min-height: 100%;
}

a {
    font-weight: bold;
    color: #F9F6F4;
}

a:hover {
    color: #F9F6F4;
    text-decoration: underline;
}

h1, h2, h3, h4, h5, h6 {
    font-family: 'Montserrat Bold';
}

hr {
    background: #2988BC;
}

/* Widths */

.w-5 {
    width: 5%;
}

.w-10 {
    width: 10%;
}

.w-20 {
    width: 20%;
}

.w-30 {
    width: 30%;
}

.w-33 {
    width: 33%;
}

.w-35 {
    width: 35%;
}

.w-40 {
    width: 40%;
}

.w-45 {
    width: 45%;
}

.w-55 {
    width: 55%;
}

.w-60 {
    width: 60%;
}
