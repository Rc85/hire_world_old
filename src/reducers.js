import { combineReducers } from 'redux';
import { ToggleMenu } from './reducers/Togglers';
import { Login } from './reducers/Auths';
import { Sectors } from './reducers/Sectors';
import { Confirmation } from './reducers/Confirmation';

export const reducers = combineReducers({
    ToggleMenu, Login, Sectors, Confirmation
});