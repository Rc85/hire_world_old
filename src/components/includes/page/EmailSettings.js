import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Alert } from '../../../actions/AlertActions';
import SubmitButton from '../../utils/SubmitButton';
import { UpdateUser, LogoutUser } from '../../../actions/LoginActions';
import fetch from 'axios';
import { LogError } from '../../utils/LogError';
import { faQuestionCircle } from '@fortawesome/pro-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Tooltip from '../../utils/Tooltip';
import InputWrapper from '../../utils/InputWrapper';
import { IsTyping } from '../../../actions/ConfigActions';

class EmailSettings extends Component {
    constructor(props) {
        super(props);

        this.state = {
            newEmail: '',
            confirmEmail: '',
            status: '',
            statusMessage: ''
        }
    }

    save() {
        this.setState({status: 'Loading'});

        fetch.post('/api/user/settings/email/change', this.state)
        .then(resp => {
            if (resp.data.status === 'success') {
                this.props.dispatch(LogoutUser());
            } else if (resp.data.status === 'error') {
                this.setState({status: '', statusMessage: ''});
            }

            this.props.dispatch(Alert(resp.data.status, resp.data.statusMessage));
        })
        .catch(err => LogError(err, '/api/user/settings/email/change'));
    }

    render() {
        return(
            <div id='email-settings' className='mb-3'>
                <div>
                    <div className='mb-3'><small>NOTE: You will be logged out and not be able to log in until you verify your new email</small></div>

                    <div className='mb-3'>
                        <div className='mb-3'>
                            <InputWrapper label='New Email'>
                                <input type='email' name='new_email' id='new-email' onChange={(e) => this.setState({newEmail: e.target.value})} value={this.state.newEmail} onFocus={() => this.props.dispatch(IsTyping(true))} onBlur={() => this.props.dispatch(IsTyping(false))} />
                            </InputWrapper>
                        </div>

                        <div className='mb-3'>
                            <InputWrapper label='Confirm Email'>
                                <input type='email' name='confirm_email' id='confirm-email' onChange={(e) => this.setState({confirmEmail: e.target.value})} autoComplete='off' value={this.state.confirmEmail} onFocus={() => this.props.dispatch(IsTyping(true))} onBlur={() => this.props.dispatch(IsTyping(false))} />
                            </InputWrapper>
                        </div>
                    </div>
                </div>

                <div className='text-right'>
                    <SubmitButton type='button' value='Save' loading={this.state.status === 'Loading'} onClick={() => this.save()} disabled={this.state.newEmail !== this.state.confirmEmail || !this.state.newEmail && !this.state.confirmEmail} />
                </div>
            </div>
        )
    }
}

export default connect()(EmailSettings);