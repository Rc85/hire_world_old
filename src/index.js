import 'babel-polyfill';
import 'normalize.css';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import './styles/vendor/bootstrap.css';
import './styles/vendor/bootstrap-grid.css';
import './styles/vendor/bootstrap-reboot.css';
import './styles/index.css';
import App from './App';
import { Provider } from 'react-redux';
import { reducers } from './reducers';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

const store = createStore(reducers, applyMiddleware(thunk));
store.subscribe(() => { console.log(store.getState()) });

ReactDOM.render(
    <Provider store={store}>
        <Router basename='/'>
            <App />
        </Router>
    </Provider>,
    document.getElementById('root')
);