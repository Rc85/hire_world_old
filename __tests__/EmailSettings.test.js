import React from 'react';
import Enzyme, { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import { UpdateUser } from '../src/actions/LoginActions';
import { User } from '../__mocks__/userMock';
import { createStore, applyMiddleware } from 'redux';
import { reducers } from '../src/reducers';
import EmailSettings from '../src/components/includes/page/EmailSettings';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([thunk]);
const store = createStore(reducers, applyMiddleware(thunk));
const user = {user: User, status: 'get session success'};

Enzyme.configure({adapter: new Adapter()});
jest.setTimeout(30000);

describe('Email settings test', () => {
    const wrapper = shallow(<EmailSettings store={store} />).dive();

    it('Matches snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Renders', () => {
        expect(wrapper.find('label').at(0).text()).toEqual('Change Email:');
        expect(wrapper.find('label').at(1).text()).toEqual('New Email:');
        expect(wrapper.find('label').at(2).text()).toEqual('Confirm Email:');
    });

    it('Dispatches user update action', async(done) => {
        const expectedAction = {
            type: 'USER_UPDATE',
            user: User
        }

        const store = mockStore(user);

        await store.dispatch(UpdateUser(User))

        expect(store.getActions()[0]).toEqual(expectedAction);
        done();
    });
});