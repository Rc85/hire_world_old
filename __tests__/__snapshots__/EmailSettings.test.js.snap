// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email settings test Matches snapshot 1`] = `
<div
  className="mb-3"
  id="email-settings"
>
  <div>
    <label>
      Change Email:
    </label>
    <div
      className="bordered-container rounded mb-3"
    >
      <div
        className="mb-3"
      >
        <label
          htmlFor="new-email"
        >
          New Email:
        </label>
        <input
          className="form-control"
          id="new-email"
          name="new_email"
          onChange={[Function]}
          type="email"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          htmlFor="confirm-email"
        >
          Confirm Email:
        </label>
        <input
          autoComplete="off"
          className="form-control"
          id="confirm-email"
          name="confirm_email"
          onChange={[Function]}
          type="email"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="text-right"
  >
    <SubmitButton
      disabled={true}
      loading={false}
      onClick={[Function]}
      type="button"
      value="Save"
    />
  </div>
</div>
`;
